common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -196,6 +232,30 @@ private boolean handleUrl(final WebView view, final String url) {
         return true;
     }
 
+    private String extractSsoNonceQueryParam(String url) {
+            URI realUrl = URI.create(url);
+            String query = realUrl.getQuery();
+            if (query == null) {
+                return null;
+            }
+
+        if(realUrl.getQuery().contains("sso_nonce=")) {
+                final Matcher matcher = SSO_PATTERN.matcher(query);
+                if(matcher.matches()) {
+                    if (matcher.groupCount() != 1) {
+                        Logger.warn(TAG, "too many sso_nonce query parameters");
+                        return null;
+                    } else {
+                        String sso_nonce = matcher.group(1);
```

Comment:
Is a regex the best way to go here? Can we use [`Uri.getQueryParameter(key)`](https://developer.android.com/reference/android/net/Uri#getQueryParameter(java.lang.String)) to do the lifting for us? I do see a comment in the javadoc re: backcompat behavior around certain chars; I'm wondering if maybe using a regex was a workaround for that....

Reply:
Whichever way we go, may be good idea to have a unit test or two covering nonce format

Reply:
Let's not use Android's Uri here....